src/Resources/public/js/searchConfigPanel.js

```diff
@@ -164,10 +178,40 @@ pimcore.bundle.advancedObjectSearch.searchConfigPanel = Class.create(pimcore.ele
             value: this.data.settings ? this.data.settings.sharedUserIds : ""
         });
 
+        this.rolesSharingField = Ext.create('Ext.form.field.Tag', {
+            name: "shared_roles",
+            width: 500,
+            fieldLabel: t("bundle_advancedObjectSearch_roles"),
+            queryDelay: 0,
+            resizable: true,
+            queryMode: 'local',
+            minChars: 1,
+            store: this.rolesStore,
+            displayField: 'label',
+            valueField: 'id',
+            forceSelection: true,
+            filterPickList: true,
+            value: this.data.settings ? this.data.settings.sharedUserIds : ""
+        });
+
+        var shareItems = [this.sharingField, this.rolesSharingField];
+
+        var user = pimcore.globalmanager.get("user");
+        if (user.admin) {
+            this.shareGlobally = new Ext.form.field.Checkbox(
+                {
+                    fieldLabel: t("share_globally"),
+                    inputValue: true,
+                    name: "share_globally",
+                    value: this.data.settings ? this.data.settings.shareGlobally : false
```

Comment:
shared globally is not initialized properly when opening saved search, even it is set in DB row